pkg/infrastructure/mongodb.go

```diff
@@ -0,0 +1,163 @@
+	MongoDBAppSecretPasswordKey = "password"
+)
+
+var (
+	// MongoDBAPIVersion refers to kafka APIVersion
+	MongoDBAPIVersion = mongodb.SchemeGroupVersion.String()
+
+	mongoDBServerGroup = mongodb.SchemeGroupVersion.Group
+)
+
+// MongoDBCredential holds the credentials information of a user into an mongoDB server
+type MongoDBCredential struct {
+	AuthDatabase string `yaml:"auth-database"`
+	Username     string `yaml:"username"`
+	Password     string `yaml:"password"`
+	Database     string `yaml:"database"`
+}
+
+// IsMongoDBAvailable checks if MongoDB CRD is available in the cluster
+func IsMongoDBAvailable(client *client.Client) bool {
+	return client.HasServerGroup(mongoDBServerGroup)
+}
+
+// IsMongoDBOperatorAvailable verify if MongoDB Operator is running in the given namespace and the CRD is available
+func IsMongoDBOperatorAvailable(cli *client.Client, namespace string) (bool, error) {
+	log.Debugf("Checking if MongoDB Operator is available in the namespace %s", namespace)
+	// first check for CRD
+	if IsMongoDBAvailable(cli) {
+		log.Debugf("MongoDB CRDs available. Checking if MongoDB Operator is deployed in the namespace %s", namespace)
+		// then check if there's an MongoDB Operator deployed
+		deployment := &v1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: MongoDBOperatorName}}
+		exists := false
+		var err error
+		if exists, err = kubernetes.ResourceC(cli).Fetch(deployment); err != nil {
+			return false, nil
+		}
+		if exists {
+			log.Debugf("MongoDB Operator is available in the namespace %s", namespace)
+			return true, nil
+		}
+	} else {
+		log.Debug("Couldn't find MongoDB CRDs")
+	}
+	log.Debugf("Looks like MongoDB Operator is not available in the namespace %s", namespace)
+	return false, nil
+}
+
+// FillMongoDBCredential fills the credential of the MongoDB server deployed with the Kogito Operator.
+// Prefilled values will be taken into account and corrected if needed.
+func FillMongoDBCredential(cli *client.Client, creds *MongoDBCredential, mongoDBInstance *mongodb.MongoDB) error {
+	usernameToSearch := DefaultMongoDBUsername
+	authDBToSearch := DefaultMongoDBAuthDatabase
+	if len(creds.Username) > 0 {
+		usernameToSearch = creds.Username
+	}
+	if len(creds.AuthDatabase) > 0 {
+		authDBToSearch = creds.AuthDatabase
+	}
+
+	user := findMongoDBUserByUsernameAndAuthDatabaseOrFirst(usernameToSearch, authDBToSearch, mongoDBInstance)
+	if user == nil {
+		return fmt.Errorf("No user found for the given MongoDB Instance")
+	}
+
+	if len(user.DB) > 0 {
+		creds.AuthDatabase = user.DB
+	}
+
+	secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: user.PasswordSecretRef.Name, Namespace: mongoDBInstance.Namespace}}
+	if exists, err := kubernetes.ResourceC(cli).Fetch(secret); err != nil {
+		return err
+	} else if exists {
+		passwordKey := DefaultPasswordSecretRef
+		if user.PasswordSecretRef.Key != "" {
+			passwordKey = user.PasswordSecretRef.Key
+		}
+		creds.Password = string(secret.Data[passwordKey])
```

Comment:
Is this safe? Where's this data will be saved? In the MongoDB CR itself?

Reply:
this should be stored into a secret for the runtime/data-index to have it via env var secret